partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -2515,6 +2524,172 @@ class PartiQLSchemaInferencerTests {
+                            TupleConstraint.Open(false),
+                            TupleConstraint.UniqueAttrs(true),
+                            TupleConstraint.Ordered
+                        )
+                    )
+                )
+            ),
+            SuccessTestCase(
+                name = "unary plus on varying numeric types including missing -- this may return missing",
+                query = """
+                    SELECT +t.a AS a
+                    FROM <<
+                        { 'a': CAST(1 AS INT8) },
+                        { 'a': CAST(1 AS INT4) },
+                        { }
+                    >> AS t
+                """.trimIndent(),
+                expected = BagType(
+                    StructType(
+                        fields = mapOf(
+                            "a" to unionOf(INT4, INT8, MISSING),
+                        ),
+                        contentClosed = true,
+                        constraints = setOf(
+                            TupleConstraint.Open(false),
+                            TupleConstraint.UniqueAttrs(true),
+                            TupleConstraint.Ordered
+                        )
+                    )
+                )
+            ),
+            SuccessTestCase(
+                name = "unary plus on varying numeric types including string -- this may return missing",
+                query = """
+                    SELECT +t.a AS a
+                    FROM <<
+                        { 'a': CAST(1 AS INT8) },
+                        { 'a': CAST(1 AS INT4) },
+                        { 'a': 'hello world!' }
+                    >> AS t
+                """.trimIndent(),
+                expected = BagType(
+                    StructType(
+                        fields = mapOf(
+                            "a" to unionOf(INT4, INT8, MISSING),
+                        ),
+                        contentClosed = true,
+                        constraints = setOf(
+                            TupleConstraint.Open(false),
+                            TupleConstraint.UniqueAttrs(true),
+                            TupleConstraint.Ordered
+                        )
+                    )
+                )
+            ),
+            SuccessTestCase(
+                name = "binary plus on varying types -- this will return missing if one of the operands is not a number",
+                query = """
+                    SELECT t.a + t.b AS c
+                    FROM <<
+                        { 'a': CAST(1 AS INT8), 'b': CAST(1.0 AS DECIMAL) },
+                        { 'a': CAST(1 AS INT4), 'b': TRUE },
+                        { 'a': 'hello world!!', 'b': DATE '2023-01-01' }
+                    >> AS t
+                """.trimIndent(),
+                expected = BagType(
+                    StructType(
+                        fields = mapOf(
+                            "c" to unionOf(INT4, INT8, MISSING, DECIMAL),
```

Comment:
Why is this returns a union of number types? 
We have `a` whose type is unionOf(INT8, INT4, String), `b` whose type is unionOf(Decimal, Bool, Date), 
Should the plus operator returns unionOf(Decimal, Missing)?

Reply:
Bool can coerce to INT4 and INT8

Reply:
Got it. But I don't think we should allow this.